### PR TITLE
Add TDRTerraformRole{Environment} to the reference generator policy

### PR DESCRIPTION
**Pull request by thanhz**

### CODE REVIEW

File: modules/environment-roles/templates/shared_reference_generator_policy.json.tpl [hunk 1]
```diff
@@ -9,7 +9,8 @@
       ],
       "Resource": [
         "arn:aws:iam::${account_id}:policy/DaReferenceCounterTableAccessPolicy${title(environment)}",
-        "arn:aws:iam::${account_id}:role/da*"
+        "arn:aws:iam::${account_id}:role/da*",
+        "arn:aws:iam::${account_id}:role/TDRTerraformRole${title(environment)}"
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_reference_generator_policy.json.tpl:13]: This should be the policy not the role that will change the policy, ie: `arn:aws:iam::${account_id}:policy/DAReferenceGeneratorLambdaPolicy${title(environment)}`

**Review by TomJKing — COMMENTED** [review]
Just one comment. Thanks.

File: modules/environment-roles/templates/shared_reference_generator_policy.json.tpl (hunk 1 above)
Comment (thanhz) [reply modules/environment-roles/templates/shared_reference_generator_policy.json.tpl]: Sorry, changed that now.
I noticed the very top policy `DaReferenceCounterTableAccessPolicy${title(environment)}` should Da be all uppercase?

Comment (thanhz) [reply modules/environment-roles/templates/shared_reference_generator_policy.json.tpl]: Changed that now. For the `DaReferenceCounterTableAccessPolicy${title(environment)}`. Should Da be all uppercase? i.e.  `DAReferenceCounterTableAccessPolicy${title(environment)}`

Comment (TomJKing) [reply modules/environment-roles/templates/shared_reference_generator_policy.json.tpl]: Yes, I'll get on to the dev who put the lowercase name in to fix it ;)

File: modules/environment-roles/templates/shared_reference_generator_policy.json.tpl [hunk 2]
```diff
@@ -9,7 +9,9 @@
       ],
       "Resource": [
         "arn:aws:iam::${account_id}:policy/DaReferenceCounterTableAccessPolicy${title(environment)}",
-        "arn:aws:iam::${account_id}:role/da*"
+        "arn:aws:iam::${account_id}:role/da*",
+        "arn:aws:iam::${account_id}:policy/DAReferenceGeneratorLambdaPolicy${title(environment)}",
+        "arn:aws:iam::${account_id}:policy/da-reference-generator-${title(environment)}-vpc-policy"
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_reference_generator_policy.json.tpl:14]: Shouldn't be using the `title` function here. Should be: arn:aws:iam::${account_id}:policy/da-reference-generator-${environment}-vpc-policy

File: modules/environment-roles/templates/shared_reference_generator_policy.json.tpl [hunk 3]
```diff
@@ -9,7 +9,9 @@
       ],
       "Resource": [
         "arn:aws:iam::${account_id}:policy/DaReferenceCounterTableAccessPolicy${title(environment)}",
-        "arn:aws:iam::${account_id}:role/da*"
+        "arn:aws:iam::${account_id}:role/da*",
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_reference_generator_policy.json.tpl:12]: Could you order the entries alphabetically.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**